Meanmodel workflow and new testcases

KatrinCoboeken opened the pull request.

KatrinCoboeken committed:
new population plotting code

KatrinCoboeken committed:
add mean model workflow.m
automatic workflow.m creation
update on tests

KatrinCoboeken committed:
Merge branch 'develop' of https://github.com/KatrinCoboeken/Reporting-Engine into develop

# Conflicts:
#	src/code/engine_library/addDosetablePerBSA.m
#	src/code/engine_library/calculateBodySurfaceArea.m
#	src/code/engine_library/getLabelWithUnit.m
#	src/code/engine_library/loadSelectedPathsOfPopulation.m
#	src/code/engine_library/plotLoopPopulationVPCPKparameter.m
#	src/code/engine_library/plotLoopPopulationVPCforPhysiology.m
#	src/code/engine_library/plotReportBoxwhisker.m
#	src/code/engine_library/plotReportTimeProfile.m
#	src/code/engine_library/readPKParameter.m
#	src/code/engine_library/removeForbiddenLetters.m
#	src/code/engine_library/textVPCPopulation.m
#	src/code/engine_library/writeTabCellArray.m
#	src/code/templates/Output.xlsx
#	test/List of Tests.xlsx
#	test/runAllTests.m
#	test/test_NO1/Input/checkTestResult.m
#	test/test_NO1/Input/workflow.m
#	test/test_NO2/Input/workflow.m
#	test/test_NO3/Input/workflow.m
#	test/test_NO3/Output/logfile.txt
#	test/test_NO3/Output/simulations/SingleIvBolus-PK-Analyses.csv
#	test/test_NO3/Output/tmp/SingleIvBolus/applicationProtocol.mat
#	test/test_NO3/Output/tmp/SingleIvBolus/outputList.mat
#	test/test_NO3/Output/tmp/SingleIvBolus/pKPList.mat
#	test/test_NO3/Output/tmp/SingleIvBolus/pop.mat
#	test/test_NO3/Output/tmp/SingleIvBolus/simResult.mat
#	test/test_NO3/Output/workflow.m
#	test/test_NO4/Input/workflow.m
#	test/test_NO5/Input/workflow.m
#	test/test_NO6/Input/workflow.m
#	test/test_NO6/Output/logfile.txt
#	test/test_NO6/Output/simulations/SingleIvBolus-PK-Analyses.csv
#	test/test_NO6/Output/tmp/SingleIvBolus/applicationProtocol.mat
#	test/test_NO6/Output/tmp/SingleIvBolus/outputList.mat
#	test/test_NO6/Output/tmp/SingleIvBolus/pKPList.mat
#	test/test_NO6/Output/tmp/SingleIvBolus/pop.mat
#	test/test_NO6/Output/tmp/SingleIvBolus/simResult.mat
#	test/test_NO6/Output/workflow.m
#	test/test_NO8/Input/workflow.m
#	test/test_NO8/Output/workflow.m

Yuri05 commented on src/code/engine_library/getDefaultVPCPopulationSettings.m:
```diff
@@ -1,89 +1,183 @@
+            'Organism|Height', 'Height', 'cm',{};...
+            'Organism|BMI', 'BMI', 'kg/m�',{};...
+            'Gender', 'Gender', '',{'Male','Female'}};
         
```
Should BSA be also included?

KatrinCoboeken replied:
I will add it for the next version

Yuri05 commented on src/code/engine_library/getDefaultVPCPopulationSettings.m:
```diff
@@ -1,89 +1,183 @@
+            'Organism|Height', 'Height', 'cm',{};...
+            'Organism|BMI', 'BMI', 'kg/m�',{};...
+            'Gender', '', '',{'Male','Female'};...
+            '<addOntogenyFactor>','','',{}};
```
Above (s. BSA question) ontogenies were not included. Is that correct?

KatrinCoboeken replied:
if the code finds the key word <addOntogenyFactor> in ths cell array then all ontogeny factors included in the xml file are added to this structure automatically during initialization of the workflow.

Yuri05 commented on src/code/engine_library/getUnitFactorForUnknownDimension.m:
```diff
@@ -27,34 +27,51 @@
         error('molweight is needed');
     end        
-    unitFactor = getUnitFactor(unit,targetUnit,unitList_dimensionList{ix_dim(1)},'MW',MW*1e9);
+    unitFactor = getUnitFactor(unit,targetUnit,unitListDimensionList{ixDim(1)},'MW',MW*1e9);
```
1e9 is factor between kg/µmol and ?

KatrinCoboeken replied:
getUnitFactor expects molweight in g/mol

Yuri05 commented on src/code/engine_library/plotReportHistogram.m:
```diff
@@ -1,74 +1,154 @@
+%         no csv
+        return
+end
+
```
why not just "if flag=='isResiduals' return"?

Yuri05 reviewed (commented):
Missing "otherwise" in many switch statements

Yuri05 commented on src/code/engine_library/readPopulationCSV.m:
```diff
@@ -7,8 +7,8 @@
+
 
 if ~exist(csvfile,'file')
     error('This file does no exist: %s')
```
I guess it should be
error('This file does no exist: %s', csvfile)

Yuri05 commented on src/code/engine_library/runMeanModelWorkflow.m:
```diff
@@ -0,0 +1,234 @@
+    for iSet = 1:length(MeanModelSet)
+        [success,VPC] = prepareMeanModelSimulation(WSettings,MeanModelSet(iSet),VPC);
+        successInputCheck = success & successInputCheck;
+    end
```
here ALL models are processed even if _prepareMeanModelSimulation_ fails on the very first model.
on purpose?

Yuri05 commented on src/code/engine_library/runPopulationSimulation.m:
```diff
@@ -84,14 +89,13 @@ function exportResult(SimResult)
 % timepoints, which happens to be not the case for simulations outside
-% PK-Sim (???), but the differences are only small % todo @Juri discuss
-% this comment with Juri
+% PK-Sim (???), but the differences are only small
```
**_What is this about??_**

Yuri05 commented on src/code/engine_library/runPopulationWorkflow.m:
```diff
@@ -79,8 +76,18 @@ function runPopulationWorkflow(TaskList,Settings,PopRunSet,varargin)
-        successInputCheck = preparePopulationSimulationConsistency(Settings,PopRunSet(iSet)) & successInputCheck;
+        [success,VPC] = preparePopulationSimulation(WSettings,PopRunSet(iSet),VPC);
+        successInputCheck = success & successInputCheck;
     end
```
Same as above: ALL simulations are processed even if it fails for one simulation. On purpose?

Yuri05 commented on src/code/engine_library/runPopulationWorkflow.m:
```diff
@@ -89,53 +96,60 @@ function runPopulationWorkflow(TaskList,Settings,PopRunSet,varargin)
-% end
+catch exception
+    
+    save('exception.mat','exception')
```
will write exception.mat in the current folder. Ok so?

Yuri05 commented on src/code/templates/prepareMeanModelWorkflow.m:
```diff
@@ -0,0 +1,137 @@
+MeanModelSet = rmfield(MeanModelSet,'outputsheet');
+
+%%write new worflow.m
+fid = fopen('workflow.m','w');
```
current folder is ok?

Yuri05 commented on src/code/templates/preparePopulationWorkflow.m:
```diff
@@ -0,0 +1,140 @@
+
+
+%%write new worflow.m
+fid = fopen('workflow.m','w');
```
current folder is ok?

Yuri05 commented on src/code/templates/readOutputXls.m:
```diff
@@ -0,0 +1,90 @@
+end
+
+% read data
+[~,~,xlsdata] = xlsread(outputXls,outputsheet);
```
We should call xlsread in 'basic' mode. (then we will directly see if it will work under Linux)

Yuri05 commented on src/code/templates/readWorkflowInput.m:
```diff
@@ -0,0 +1,106 @@
+
+
+% read data
+[~,~,xlsdata] = xlsread(workflowInputxls,sheet);
```
We should call xlsread in 'basic' mode. (then we will directly see if it will work under Linux)

KatrinCoboeken replied:
functions in template directory, are not planned to run under Linux, they are executed "before" starting the workflow.m

KatrinCoboeken committed:
new if nInd > 1000 save simulation results in bunches
parallelisation (use same structurs and functions) of Individual simulation and Population simulation
